node/src/query.ts

```diff
@@ -129,7 +129,7 @@ export class Query<T = number[]> {
       const newObject: Record<string, unknown> = {}
       Object.keys(entry).forEach((key: string) => {
         if (entry[key] instanceof Vector) {
-          newObject[key] = (entry[key] as Vector).toArray()
+          newObject[key] = (entry[key] as Vector).toJSON()
```

Comment:
Why this change?

Reply:
Arrow `toArray()` returns f16 array as uint16

Reply:
That seems worth a comment.

Reply:
done